### PR TITLE
Add the ability to specify attributes on HDF5Datasets

### PR DESCRIPTION
**Pull request by DominicOram**
See https://github.com/DiamondLightSource/ADOdin/issues/22

To test:
* Run unit tests and confirm they pass

### CODE REVIEW

**Review by GDYendell — APPROVED** [review]
This looks great. Only nitpicks.

**Comment by DominicOram** [comment]
Thanks for the review, comments addressed

File: python/src/odin_data/meta_writer/hdf5dataset.py [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+from typing import Optional
 
 import h5py as h5
 import numpy as np
 
 
+def units(unit: str):
+    return {"units": unit}
+
+
```
Comment (ajgdls) [inline python/src/odin_data/meta_writer/hdf5dataset.py:12]: Where is this used?

Comment (DominicOram) [reply python/src/odin_data/meta_writer/hdf5dataset.py:12]: In the detector specific code at https://github.com/DiamondLightSource/eiger-detector/pull/24/files. The use of `units` as an attribute name seemed useful enough that it should live in the core writer. Happy to move it into the `eiger` if you'd rather though

**Review by ajgdls — APPROVED** [review]

**Pull request merged**